zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java

```diff
@@ -141,12 +144,12 @@ public boolean hasPermission(String name, String action, Object... arg) {
         if (result) {
             if (log.isDebugEnabled()) {
                 log.debug("ALLOWED hasPermission({}, {}, {}) for user {}",
-                        name, action, arg, getAccountUsername());
+                        name, action, Lists.newArrayList(arg), getAccountUsername());
             }
         } else {
             if (log.isDebugEnabled()) {
```

Comment:
I think originally it was to prevent the getAccountUser() method call if it's not in debug mode. Now it will also prevent the creation of the list.